src/client/options/mod.rs

```diff
@@ -1372,11 +1408,22 @@ impl ClientOptionsParser {
                 self.zlib_compression = Some(i);
             }
 
-            _ => {
-                return Err(ErrorKind::ArgumentError {
-                    message: "invalid option warning".to_string(),
+            other => {
+                let (jaro_winkler, option) = URI_OPTIONS.iter().fold((0.0, ""), |acc, option| {
```

Comment:
I think this can be written a bit more concisely by using [`Iterator::max_by`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.max_by).

Reply:
I re-implemented this work using `max_by` but I'm not sure if it's better than using fold. Here is what the new call would look like:
```
let option = URI_OPTIONS.iter().max_by(|o1, o2| {
    if jaro_winkler(other, o1).abs() > jaro_winkler(other, o2).abs() {
        return Ordering::Greater;
    } else {
        return Ordering::Less;
    }
});
let similarity = jaro_winkler(other, option.unwrap_or(&"")).abs();
```
A few downsides here: since the `f64` type doesn't implement `cmp()` the comparison logic needs to be done manually, and the score needs to be re-calculated at the end because `max_by` only returns the winning string. It does look a little cleaner, though. Would you consider this to be a stylistically better implementation?

Reply:
Ah, I had forgotten that `f64` doesn't implement `cmp`. I think you can get around the recalulation by calling `map` before doing `max_by`, though.